sympy/physics/mechanics/tests/test_kane.py

```diff
@@ -74,8 +78,31 @@ def test_two_dof():
     assert expand(rhs[1]) == expand((k1 * q1 + c1 * u1 - 2 * k2 * q2 - 2 *
                                     c2 * u2) / m)
 
-    assert simplify(KM.rhs() -
-                    KM.mass_matrix_full.LUsolve(KM.forcing_full)) == zeros(4, 1)
+    # Check that the explicit form is the default and kinematic mass matrix is identity
+    assert KM.explicit_kinematics
+    assert KM.mass_matrix_kin == eye(2)
+
+    # Check that the implicit and explicit matrices are different when involving kinematics
+    mf_keys = ['mass_matrix', 'mass_matrix_kin', 'mass_matrix_full',
+               'forcing', 'forcing_kin', 'forcing_full']
+    KM_matrices = {}
+    for explicit_kinematics in [False, True]:
+        KM.explicit_kinematics = explicit_kinematics
+        KM_matrices['explicit' if explicit_kinematics else 'implicit'] = {k:KM.__getattribute__(k) for k in mf_keys}
+
+    for k in mf_keys:
+        if k in ['mass_matrix', 'forcing']:
+            # dynamic matrix/vector are unchanged whether we use explicit or implicit kinematics
+            assert KM_matrices['explicit'][k] == KM_matrices['implicit'][k]
+        else:
+            # but matrix/vector involving kinematics are expected to be different
+            assert KM_matrices['explicit'][k] != KM_matrices['implicit'][k]
+
+    # Check that whether using implicit or explicit kinematics the RHS equations are consisten with the matrix form
+    for explicit_kinematics in [False, True]:
+        KM.explicit_kinematics = explicit_kinematics
+        assert simplify(KM.rhs() -
+                        KM.mass_matrix_full.LUsolve(KM.forcing_full)) == zeros(4, 1)
```

Comment:
I would personally not usethese kind of test loops, since they are not nice to debug. Rather write them out, since you don't even use less lines in the end. Besides that I would also suggest to directly check the implicit kinematic mass matrix and forcing vector (they are really simple), i.e. `KM.mass_matrix_kin == Matrix([[1/2, 0], [0, 2]])` (is also bit more useful and accurate)

P.S. Try to keep the lines within 80 characters

Reply:
Let's just try to get this in. We've nitpicked @zouhairm enough :)

Reply:
- the for loop was on purpose, not to save lines of code, but to make it clear that the only thing changing between the two tests is the explicit flag. pytest should print out the variables when something fails, so hopefully shouldn't be too difficult to debug.

-  I've removed one of them anyway by implementing a more, excuse the pun, explicit matrix check as you suggested as opposed to just checking that they are different

- sorry about the 80 chars: I know it's a PEP8 thing but I didn't see it enforced by code_quality so I assumed the codebase uses a different value (we certainly do in ours as 80 is just too small for most modern monitors). If you feel strongly about it lmk and I'll clean it up.

Reply:
I do agree with the first point, was just a bit surprised to see this kind of code in tests. Furthermore, thanks for taking the time to update. As for the last point, the 80 character line width is in SymPy as far as I know not a strict rule, but personally I like consistency in formatting (even though I personally prefer 120 chars).